graders/java/Dockerfile

```diff
@@ -17,13 +17,15 @@ RUN groupadd sbuser \
     # Disable ptrace mechanism to avoid heap dumping from other processes
     && echo "kernel.yama.ptrace_scope = 3" >> /etc/sysctl.d/10-ptrace.conf \
     # Delete all java.management files to hinder student's ability to retrieve information from the grader
-    && rm -rf /usr/lib/jvm/java-21-openjdk-amd64/lib/libmanagement.so /usr/lib/jvm/java-21-openjdk-amd64/lib/libmanagement_agent.so /usr/lib/jvm/java-21-openjdk-amd64/lib/libmanagement_ext.so \
-    mkdir /javagrader
+    && rm -rf /usr/lib/jvm/java-25-openjdk-*/lib/libmanagement.so /usr/lib/jvm/java-25-openjdk-*/lib/libmanagement_agent.so /usr/lib/jvm/java-25-openjdk-*/lib/libmanagement_ext.so
 
 COPY libs /javagrader/libs
 COPY JUnitAutograder.java AutograderInfo.java /javagrader/
 
-RUN javac -cp '/javagrader:/javagrader/libs:/javagrader/libs/*' -d /javagrader /javagrader/JUnitAutograder.java /javagrader/AutograderInfo.java
+# Compile the autograder with all warnings enabled, except for serial (missing
+# serialVersionUID) and this-escape (use `this` reference in constructor)
+# warnings
+RUN javac -Xlint:all -Xlint:-serial -Xlint:-this-escape -Werror -cp '/javagrader:/javagrader/libs:/javagrader/libs/*' -d /javagrader /javagrader/JUnitAutograder.java /javagrader/AutograderInfo.java
```

Comment:
The flags were added mainly to add an extra layer of verification of the autograder code.